modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2651,30 +2654,40 @@ def from_pandas(cls, df):
         # it into columns either because, otherwise, we don't know
         # the number of rows and, thus, unable to restore the index.
         # That's what we usually have for arrow tables and execution
-        # result. Unnamed index is renamed to __index__. Also all
+        # result. Unnamed index is renamed to {IDX_COL_PREF}. Also all
         # columns get 'F_' prefix to handle names unsupported in HDK.
-        elif len(new_index) == 0 or (
-            len(new_columns) != 0 and cls._is_trivial_index(new_index)
-        ):
+        elif (
+            len(new_index) == 0
+            and not isinstance(new_index, MultiIndex)
+            and new_index.name is None
+        ) or (len(new_columns) != 0 and cls._is_trivial_index(new_index)):
             index_cols = None
         else:
             orig_index_names = new_index.names
             orig_df = df
-
             index_cols = cls._mangle_index_names(new_index.names)
             df.index.names = index_cols
             df = df.reset_index()
-
             orig_df.index.names = orig_index_names
+
         new_dtypes = df.dtypes
-        df = df.add_prefix("F_")
+
+        def encoder(n):
+            return n if n == MODIN_UNNAMED_SERIES_LABEL else encode_col_name(n, False)
```

Comment:
could you please pass this as a kwarg so it would read better
```suggestion
            return n if n == MODIN_UNNAMED_SERIES_LABEL else encode_col_name(n, ignore_reserved=False)
```